src/types/redis_stream_base.h

```diff
@@ -239,6 +239,36 @@ struct StreamAutoClaimResult {
   std::vector<std::string> deleted_ids;
 };
 
+struct StreamPendingOptions {
+  uint64_t idle_time = 0;
+  bool with_time = false;
+
+  StreamEntryID start_id{StreamEntryID::Minimum()};
+  StreamEntryID end_id{StreamEntryID::Maximum()};
+
+  uint64_t count;
+  bool with_count = false;
+  bool with_consumer = false;
+
+  std::string consumer;
+  std::string stream_name;
+  std::string group_name;
+};
+
+struct StreamGetPendingEntryResult {
+  uint64_t pending_number;
+  StreamEntryID smallest_id;
+  StreamEntryID greatest_id;
+  std::vector<std::pair<std::string, int>> consumer_infos;
+};
+
+struct StreamGetExtPendingEntryResult {
```

Comment:
Redis named as `StreamNACK` which seems a more reasonable name for the message queue: https://github.com/redis/redis/blob/a9267137ee5cbd0908d3844e9d57284e93d85b72/src/t_stream.c#L3025